beacon-chain/state/stategen/getter.go

```diff
@@ -56,7 +56,11 @@ func (s *State) StateByRoot(ctx context.Context, blockRoot [32]byte) (state.Beac
 
 	// Genesis case. If block root is zero hash, short circuit to use genesis state stored in DB.
 	if blockRoot == params.BeaconConfig().ZeroHash {
-		return s.beaconDB.GenesisState(ctx)
+		root, err := s.beaconDB.GenesisBlockRoot(ctx)
```

Comment:
Hello Author.

The change here is changing the semantics of the function. 

* New change is using `GenesisBlockRoot`: won't allow empty record
* Original code is using `GensisState`: allow empty record, and won't return error.  https://github.com/prysmaticlabs/prysm/blob/77d3ccb9ad8b28656452456d2d25437f5fc53d88/beacon-chain/db/kv/state.go#L91

I think this change will cause trouble when a fresh start of the chain (no `genesis-root` written yet)

Reply:
By right we should always have a genesis state, does this change break anything for you ?

Reply:
- Yes, there is always a `genesis state`.   The new patch assume that this genesis state is stored in DB already. 

- Yes, could you try to bring up a "refresh" node?  Such as `interop`, it should prevent the node to start.